### DISCUSSION TITLE
Zones and intersection

### DISCUSSION THREAD

**Discussion by svk-man** [alexbol99/flatten-js#134] [Q&A]
![image](https://user-images.githubusercontent.com/17108861/208332313-fcb8558f-d47e-4e1d-abd5-352b2a041ebf.png)

Good afternoon!

I need to draw three polygons - outer and two inner ones. They can be edited.

All three polygons form zones. I marked them - 1, 2, 3.

I need to be able to switch between zones and look for the intersection of each of the zones with a moving point, for example, the mouse cursor.

Can you please tell me if your library will help me?

**Comment by alexbol99** [comment]
Hi, @svk-man ,

If you need to catch mouse events, you can use flatten-js together with s3, see notebook:
https://observablehq.com/@alexbol99/zones-and-intersection-39

If you question is about zone calculation, please give more details.

Best,
Alex
